### ISSUE TITLE
Problems using panel for searching for all cases for institute, when it exists with same name for other institute?

### ISSUE THREAD

**Issue by dnil** [Clinical-Genomics/scout#3915]
There appear to be some issues with searching with a panel when multiple ones with the same `panel_name` exist for different insitutes.

Users report not being able to see either, and as an admin I get both when on just one of the institutes. 

![Screenshot 2023-04-26 at 11 15 20](https://user-images.githubusercontent.com/758570/234532805-acced415-87f5-4f19-a732-d49d0b608015.png)

Guessing we have some issues not checking both `institute` and `panel_name` plus `version` when selecting where we don't have the actual panel `_id`. We could either fix that lookup, or as a workaround (which probably would be helpful to avoid versioning confusion) not allow users to assign already taken panel names.

**Comment by northwestwitch** [comment]
Perhaps we can include also a fix to this bug in the upcoming release? I can work on it today!

**Comment by dnil** [comment]
Great, agreed! I have a few meetings anyway, so aiming for a release tomorrow sounds like a plan.

**Comment by northwestwitch** [comment]
What page is this?

<img width="688" alt="image" src="https://user-images.githubusercontent.com/28093618/234841505-b5e900de-d0cb-454b-891d-33346523dde8.png">

**Comment by northwestwitch** [comment]
Ah I got it! cases, as in the title! :)

**Comment by northwestwitch** [comment]
Mmm that's weird, you are not even supposed to have 2 panels with the same name in the database..

<img width="1150" alt="image" src="https://user-images.githubusercontent.com/28093618/234845612-d1e757e4-ac1b-4b9e-a75b-46d32b8019d5.png">

**Comment by dnil** [comment]
that's weird! I tried as well, with a new institute and got the same. There are no actual limits for the db, but seems someone has been a bit creative to get this to happen!

**Comment by northwestwitch** [comment]
I think it was possible to create 2 panels with the same name in the past, we must have changed it at some point. Good, so one thing is already fixed.

**Comment by northwestwitch** [comment]
Ok, I can't reproduce it locally?

- create a new institute: `scout --demo load institute -i cust00`
- Create a new panel for this institute (you are not allowed to name it panel1, so name it panel2)
- From inside MongoDB compass, rename the new panel to `panel1`, same as the demo panel already present
- Edit the config file of the demo case 643595 so it's belonging to cust001
- Load this other demo case: `scout --demo load case scout/demo/643595.config.yaml`

As an admin, when on cases page of cust000, search for cases with panel1:

I get only one case, from cust000

Same if I go on cases page of cust001, I get only one case from cust001

**Comment by dnil** [comment]
🕸️ 
<img width="380" alt="Screenshot 2023-04-27 at 13 45 04" src="https://user-images.githubusercontent.com/758570/234852512-2dd98d61-5528-4f1a-84c7-502bef1d15b5.png">

**Comment by northwestwitch** [comment]
> 🕸️ <img alt="Screenshot 2023-04-27 at 13 45 04" width="380" src="https://user-images.githubusercontent.com/758570/234852512-2dd98d61-5528-4f1a-84c7-502bef1d15b5.png">

HOW???

**Comment by dnil** [comment]
Hm, I can ask her: maybe she had help..

**Comment by dnil** [comment]
On careful inspection the panel names are actually slightly different, although they begin the same. 🙄

So I checked with Maria and after some explanation, we agree she has only the same "issue" I have, seeing the panel twice in the list. Perhaps this means the drop down is from the user institutes panels, not from the current edited settings institute panels.

So, agreed on no need to change the panel edit part, but perhaps still the institute settings?

**Comment by northwestwitch** [comment]
> . Perhaps this means the drop down is from the user institutes panels, not from the current edited settings institute panels.
> 
> So, agreed on no need to change the panel edit part, but perhaps still the institute settings?

While I now understand how this happened, I don't get these final sentences. Which dropdown, the one for searching cases? Could you explain what we need to fix?

**Comment by dnil** [comment]
Ah, great! It took me a long time to realise the institutes that had the panels with the same display name, and similar panel names were collaborators, and because of https://github.com/Clinical-Genomics/scout/blob/5168094a4c7fc06669c7ab3c410f5469d3d477c7/scout/server/blueprints/institutes/controllers.py#L267 they are both shown. It would not hurt to block also having the same display name for panels, as content can be different, but as long as we understand it's all fine. I think we can close it and leave it to the users if they want to risk having the same name for two collaborator institutes.

**Comment by northwestwitch** [comment]
> Ah, great! It took me a long time to realise the institutes that had the panels with the same display name, and similar panel names were collaborators, and because of
> 
> https://github.com/Clinical-Genomics/scout/blob/5168094a4c7fc06669c7ab3c410f5469d3d477c7/scout/server/blueprints/institutes/controllers.py#L267
> 
> they are both shown. It would not hurt to block also having the same display name for panels, as content can be different, but as long as we understand it's all fine. I think we can close it and leave it to the users if they want to risk having the same name for two collaborator institutes.

Ahhhh I see! It was a tricky one! They could name it panelxxx_custyyy or something like this?

**Comment by northwestwitch** [comment]
Or we could fix it so that collaborators have direct access to the other institute's panels? I they don't have it already?

**Comment by dnil** [comment]
Yes, or just accept that they can for the most part do everything they need with the panel if they maintain it on one institute, and share it with their collaborators through that mechanism. Not sure there are any scenarios when they really, really need duplicates, and also can collaborate?
